internal/discovery/services_discoverer.go

```diff
@@ -34,20 +49,9 @@ func (d *servicesDiscoverer) Discover() ([]*corev1.Service, error) {
 
 func NewServicesDiscoverer(client kubernetes.Interface) ServiceDiscoverer {
 	// Arbitrary value, same used in Prometheus.
-	resyncDuration := 10 * time.Minute
-	stopCh := make(chan struct{})
-	sl := func(options ...informers.SharedInformerOption) ServicesLister {
-		factory := informers.NewSharedInformerFactoryWithOptions(client, resyncDuration, options...)
-
-		lister := factory.Core().V1().Services().Lister()
-
-		factory.Start(stopCh)
-		factory.WaitForCacheSync(stopCh)
-
-		return lister
-	}
+	sl, _ := NewServicesLister(client)
```

Comment:
Does it make sense to keep exposing the services discoverer if it does exactly the same of the lister?

Reply:
Probably not, since we are not using it anywhere. But as it might be useful during the refactor I did not want to remove it.
I think we can garbage-collect it if we haven't found a use case at the end of the epic.